File: test_project/sample.ts
// Used function
export function usedFunction() {
    console.log("This is used - updated!");
    helperFunction();
}

// Helper function (used internally)
function helperFunction() {
    console.log("Helper");
}

// Dead code - not used anywhere
// @ts-ignore - index-chan: Test file - may be used in tests
function deadFunction() {
    console.log("This is dead code");
}

// Dead code - exported but not used
export function unusedExport() {
    console.log("Exported but unused");
}

// Class with dead methods
class MyClass {
    usedMethod() {
        console.log("Used");
    }
    
// @ts-ignore - index-chan: Test file - may be used in tests
    deadMethod() {
        console.log("Dead");
    }
}

const instance = new MyClass();
instance.usedMethod();

File: test_rust_project/sample.rs
// Test Rust file for index-chan

pub fn used_function() {
    println!("This function is used");
    helper_function();
}

fn helper_function() {
    println!("Helper function");
}

#[allow(dead_code)] // index-chan: Test file - may be used in tests
#[allow(dead_code)] // index-chan: Test file - may be used in tests
#[allow(dead_code)] // index-chan: Test file - may be used in tests
#[allow(dead_code)] // index-chan: Test file - may be used in tests
fn unused_function() {
    println!("This function is never called");
}

#[allow(dead_code)] // index-chan: Test file - may be used in tests
#[allow(dead_code)] // index-chan: Test file - may be used in tests
#[allow(dead_code)] // index-chan: Test file - may be used in tests
#[allow(dead_code)] // index-chan: Test file - may be used in tests
pub fn another_used() {
    used_function();
}
#[allow(dead_code)] // index-chan: Test file - may be used in tests
#[allow(dead_code)] // index-chan: Test file - may be used in tests
#[allow(dead_code)] // index-chan: Test file - may be used in tests
#[allow(dead_code)] // index-chan: Test file - may be used in tests

fn dead_code_example() {
    // This is dead code
    let x = 42;
}